kv/etcd/etcd.go

```diff
@@ -87,6 +102,10 @@ func (c *client) Get(key string) (kv.Value, error) {
 	r, err := c.kv.Get(ctx, c.opts.KeyFn()(key))
 	if err != nil {
 		c.m.etcdGetError.Inc(1)
+		cachedV, ok := c.getCache(key)
```

Comment:
The expected behavior should be documented, i.e., we favor remote reads and only fall back to in-memory cache if the storage layer is unavailable